From 762891e5a5a21f9e38b1957a270254d17cd5a28d Mon Sep 17 00:00:00 2001
From: Alexey Chekmarev <alexey.chekmarev@intel.com>
Date: Wed, 30 Sep 2015 10:47:15 +0200
Subject: [PATCH] fdp_nfc: fix recovery mechanism issue in FdP driver

the driver mutex must be unlocked before leaving the driver after
h/w reset execution

CQ: MWG100247419

Change-Id: Idbfdb88e46795dcb30949d9d9424fc366a0d98fd
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-4643
Signed-off-by: Sebastian Niciarz <sebastian.niciarz@intel.com>
---
 drivers/staging/fdp/fdp_custom.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/fdp/fdp_custom.c b/drivers/staging/fdp/fdp_custom.c
index 0d2c550..4ebc48e 100644
--- a/drivers/staging/fdp/fdp_custom.c
+++ b/drivers/staging/fdp/fdp_custom.c
@@ -551,7 +551,7 @@ static void fdp_irqout_read(struct fdp_custom_device *p_device)
 			{
 				/* seems we enter in an unrecoverable error */
 				fdp_reset(p_device);
-				return;
+				break;
 			}
 
 			continue;
-- 
1.9.1

